src/app/_components/CategorySelect.tsx

```diff
@@ -35,36 +37,65 @@ export function CategorySelect({
         touchTarget.offsetClassY,
       )}
     >
+      {children}
+    </ul>
+  )
+}
+
+export function CategorySelect({
+  categoryOption,
+  categorySettings,
+  categoryCounts,
+  onCategoryOptionChange,
+}: CategorySelectProps) {
+  return (
+    <CategoryWrapper>
       {categorySettings.map((option) => {
         const isSelected = categoryOption === option.id
 
         const countOrUndefined = categoryCounts
           ? categoryCounts[option.id] || 0
           : undefined
-        const hasCount = typeof countOrUndefined === 'number'
 
         return (
-          <li key={option.id}>
-            <button
-              className={clsx(
-                'text-pretty rounded-lg py-2 text-left font-bold focus:brand-outline hover:bg-brand-700',
-                hasCount && 'inline-flex items-baseline gap-2',
-                touchTarget.class,
-                {
-                  'bg-brand-700 text-brand-400': isSelected,
-                  'bg-transparent text-brand-300': !isSelected,
-                },
-              )}
-              onClick={() => onCategoryOptionChange(option.id)}
-            >
-              <span>{option.name}</span>
-              {hasCount && (
-                <span className="text-sm font-light">({countOrUndefined})</span>
-              )}
-            </button>
-          </li>
+          <CategoryListItem
+            key={option.id}
+            name={option.name}
+            isSelected={isSelected}
+            count={countOrUndefined}
+            handleClick={() => onCategoryOptionChange(option.id)}
+          />
         )
       })}
-    </ul>
+    </CategoryWrapper>
+  )
+}
+
+export function CategoryListItem({
```

Comment:
If we have 3 components in the same file, they should be namespaced like we do in `Card` or `FilterContainer`, for readability :)

Reply:
Done.